examples/deployments/oss.py

```diff
@@ -0,0 +1,24 @@
+"""
+Usage
+-----
+mlflow deployments start-server --config-path examples/gateway/openai/config.yaml
+python examples/deployments/oss.py
+-----
+"""
+from mlflow.deployments import get_deploy_client
+
+
+def main():
+    client = get_deploy_client("http://localhost:5000")
+    print(client.list_endpoints())
+    print(client.get_endpoint(endpoint="chat"))
+    print(
+        client.predict(
+            endpoint="chat",
+            inputs={"messages": [{"role": "user", "content": "Hello!"}]},
+        ),
+    )
```

Comment:
This doesn't seem to be working for me.

```
$ python oss.py
[Endpoint(name='chat', endpoint_type='llm/v1/chat', model=RouteModelInfo(name='gpt-3.5-turbo', provider='openai'), endpoint_url='http://localhost:8000/gateway/chat/invocations'), Endpoint(name='completions', endpoint_type='llm/v1/completions', model=RouteModelInfo(name='gpt-3.5-turbo', provider='openai'), endpoint_url='http://localhost:8000/gateway/completions/invocations'), Endpoint(name='embeddings', endpoint_type='llm/v1/embeddings', model=RouteModelInfo(name='text-embedding-ada-002', provider='openai'), endpoint_url='http://localhost:8000/gateway/embeddings/invocations')]
name='chat' endpoint_type='llm/v1/chat' model=RouteModelInfo(name='gpt-3.5-turbo', provider='openai') endpoint_url='http://localhost:8000/gateway/chat/invocations'
Traceback (most recent call last):
  File "/Users/corey.zumar/mlflow/mlflow/utils/request_utils.py", line 32, in augmented_raise_for_status
    response.raise_for_status()
  File "/Users/corey.zumar/miniconda3/envs/default/lib/python3.10/site-packages/requests/models.py", line 1021, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 404 Client Error: Not Found for url: http://localhost:8000/invocations

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/corey.zumar/mlflow/examples/deployments/oss.py", line 24, in <module>
    main()
  File "/Users/corey.zumar/mlflow/examples/deployments/oss.py", line 16, in main
    client.predict(
  File "/Users/corey.zumar/mlflow/mlflow/deployments/mlflow/__init__.py", line 165, in predict
    return self._call_endpoint(
  File "/Users/corey.zumar/mlflow/mlflow/deployments/mlflow/__init__.py", line 113, in _call_endpoint
    augmented_raise_for_status(response)
  File "/Users/corey.zumar/mlflow/mlflow/utils/request_utils.py", line 35, in augmented_raise_for_status
    raise HTTPError(
requests.exceptions.HTTPError: 404 Client Error: Not Found for url: http://localhost:8000/invocations. Response text: {"detail":"Not Found"}
```

Seems like we're constructing the endpoint URL for `predict` incorrectly. Should be `http://localhost:8000/chat/invocations` but we're constructing `http://localhost:8000/invocations`

Can we fix and add a test?

Reply:
We need to strip `/` before joining paths.

Reply:
Fixed